### PR TITLE
Feature: Global Destruction

### PR DESCRIPTION
**Pull request by LindonAliu**
Closes: #782

### CODE REVIEW

File: test/functional/run/global_destructor.glu [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// RUN: gluc %s -o %t && %t | FileCheck %s
```
Comment (Snowy1803) [inline test/functional/run/global_destructor.glu]: Add a test where the global is never initialized (shouldn't be destroyed), and a test for the above edge case

**Review by Snowy1803 — COMMENTED** [review]
lgtm otherwise

**Review by Snowy1803 — APPROVED** [review]
